mm2src/lp_ordermatch.rs

```diff
@@ -1378,8 +1380,8 @@ impl<'a> MakerOrderBuilder<'a> {
         self
     }
 
-    /// Validate fields and build
-    fn build(self) -> Result<MakerOrder, MakerOrderBuildError> {
+    /// Validate fields
+    fn validate(self) -> Result<MmNumber, MakerOrderBuildError> {
```

Comment:
What do you think about adding the following?
```rust
fn validate_min_vol(...)
fn validate_max_base_vol(...)
etc
```
They may not even belong to `impl MakerOrderBuilder`. We can use them separately in `update_maker_order` and `MakerOrderBuilder::validate`. It might be not required to run all the validations on the update. Also, `self.base_coin.ticker() == self.rel_coin.ticker()` check is not required on update as order is already created with different base and rel.

We might also avoid cloning of entire builder by using reference and cloning only the required values in case of returning the error.